dttools/src/jx_sub.c

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright (C) 2021- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include "jx_sub.h"
+
+#include <assert.h>
+#include <string.h>
+
+
+static struct jx * jx_sub_call( struct jx *func, struct jx *args, struct jx *ctx) {
+    assert(func);
+    assert(args);
+    assert(jx_istype(args, JX_ARRAY));
+    assert(jx_istype(func, JX_SYMBOL));
+
+    struct jx *left  = jx_copy(func);
+    struct jx *right = NULL;
+
+    if (!strcmp(func->u.symbol_name, "range")    ||
+        !strcmp(func->u.symbol_name, "format")   ||
+        !strcmp(func->u.symbol_name, "join")     ||
+        !strcmp(func->u.symbol_name, "ceil")     ||
+        !strcmp(func->u.symbol_name, "floor")    ||
+        !strcmp(func->u.symbol_name, "basename") ||
+        !strcmp(func->u.symbol_name, "dirname")  ||
+        !strcmp(func->u.symbol_name, "listdir")  ||
+        !strcmp(func->u.symbol_name, "escape")   ||
+        !strcmp(func->u.symbol_name, "template") ||
+        !strcmp(func->u.symbol_name, "len")      ||
+        !strcmp(func->u.symbol_name, "fetch")    ||
+        !strcmp(func->u.symbol_name, "schema")   ||
+        !strcmp(func->u.symbol_name, "like")     ||
+        !strcmp(func->u.symbol_name, "keys")     ||
+        !strcmp(func->u.symbol_name, "values")   ||
+        !strcmp(func->u.symbol_name, "items")) {
+        
+        right = jx_sub(args, ctx);
+
+    } else if (!strcmp(func->u.symbol_name, "select") ||
```

Comment:
Right, there are two categories of functions: eager and lazy evaluation.  I guess we need **two** functions to match names...